File: 卡码/代码随想录/Go/哈希/两数之和/LeetCode_0001.go
//1. 两数之和
func twoSum(nums []int, target int) []int {
	hash:=make(map[int]int)
	for i,num:=range nums {
		//判断是否有两数之和等于target
		if index,ok:=hash[target-num];ok {
			return []int{index,i}
		}
		//将当前数字添加到map中
		hash[num]=i
	}
	return []int{}
}

File: 卡码/代码随想录/1/03-哈希表/Js/04-两数之和/LeetCode_0001.js
//1. 两数之和
const twoSum=(nums,target)=>{
  let map=new Map();
  for(let index in nums){
    if(map.has(target-nums[index])){
      console.log(map);
      return [index,map.get(target-nums[index])];
    }
    else map.set(nums[index],index);
  }
  return [];
}